spikeinterface/sortingcomponents/waveforms/pca.py

```diff
@@ -0,0 +1,175 @@
+
+        {}
+        
+        Returns
+        -------
+        IncrementalPCA: An incremental PCA estimator from scikit-learn.
+            The pca model
+        """
+        
+        # Detect peaks and sub-sample them
+        peaks = detect_peaks(recording, **detect_peaks_params, **job_kwargs)
+        peaks = select_peaks(peaks, **peak_selection_params) # How to select n_peaks
+
+        # Create a waveform extractor
+        ms_before = self._kwargs["ms_before"]
+        ms_after = self._kwargs["ms_after"]
+        # Creates a numpy sorting object where the spike times are the peak times and the unit ids are the peak channel
+        sorting = NumpySorting.from_peaks(peaks, sampling_frequency=recording.sampling_frequency) 
+        we = extract_waveforms(recording, sorting, ms_before=ms_before, ms_after=ms_after, folder=None, 
+                               mode="memory", max_spikes_per_unit=None, **job_kwargs)
+
+        # compute PCA by_channel_global (with sparsity)
+        sparsity = ChannelSparsity.from_radius(we, radius_um=self.local_radius_um)
+        pc = compute_principal_components(we, n_components=n_components,  mode='by_channel_global',
+                                        sparsity=sparsity, whiten=whiten)
+        self.pca_model  = pc.get_pca_model()
+
+        # model_folder should be an input
+        pca_model_path = self._kwargs["model_path"]
+
+        with open(pca_model_path, "wb") as f:
+            pickle.dump(self.pca_model, f)
+            
+        return self.pca_model
+    
+PCBaseNode.fit.__doc__ = PCBaseNode.fit.__doc__.format(_shared_job_kwargs_doc)
+class PCAProjection(PCBaseNode):
+    """
+    A step that performs a PCA projection on the waveforms extracted by a peak_detection function.
+        
+    This class can work in two ways:
+    1) The class can receive the path of a previously trained model with the `model_path` argument.
+    2) The class can be trained with the fit method after instantiating the class. The model will be stored in 
+    `model_path`.
+
+
+    Parameters
+    ----------
+    recording : BaseRecording
+        The recording object.
+    parents: list
+        The parent nodes of this node. This should contain a mechanism to extract waveforms.
+    ms_before : float, optional
+        The number of milliseconds to include before the peak of the spike, by default 1.
+    ms_after : float, optional
+        The number of milliseconds to include after the peak of the spike, by default 1.
+    peak_sign : str, optional
+        The sign of the peak, either 'neg' or 'pos', by default 'neg'.
+    all_channels : bool, optional
+        Whether to extract spikes from all channels or only the channels with spikes, by default True.
+    model_path : str, optional
+        The path to the pca model, by default None.
+    local_radius_um : float, optional
+        The radius (in micrometers) to use for definint sparsity, by default None.
```

Comment:
So this is only used for fitting. IMO, it the parent waveforms are already sparse, this should be ignored and sparsity should be inferred from the parent. We should make sure that the docs reflect this. 

A similar argument could be done for `ms_before`, `ms_after`. Even if the cut outs are not exactly the same for the parent waveforms, as long as they are larger than the cutouts the model needs, the `compute_buffer` should take care of futher clipping the waveforms. 
E.g., if the parent waveforms have `ms_before/after=2` and the PCA node has `ms_before/after=1`, at compute time the original waveforms sould be cutout to the correct length before running through the model. What do you think?

Reply:
Yes, @samuelgarcia was suggestig getting rid of all of these parameters at the init. At this moment I agree with this. I will think about it and we can discuss this today later at the meeting.